From 197426d1d916641c196cfda4530ad0e3bbcf1ac7 Mon Sep 17 00:00:00 2001
From: Ivan Vecera <ivecera@redhat.com>
Date: Mon, 8 Sep 2014 16:13:52 -0400
Subject: [ethernet] be2net: remove unused code in be_cmd_vlan_config()

Message-id: <1410192881-30702-30-git-send-email-ivecera@redhat.com>
Patchwork-id: 92455
O-Subject: [RHEL7.1 PATCH 29/78] be2net: remove unused code in be_cmd_vlan_config()
Bugzilla: 1088575
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>
RH-Acked-by: Nikolay Aleksandrov <nikolay@redhat.com>

BZ: #1088575

Upstream commit(s):
commit 4d567d9734ea83ec3dabeb29321584da5f6f0b6c
Author: Kalesh AP <kalesh.purayil@emulex.com>
Date:   Fri May 9 13:29:17 2014 +0530

    be2net: remove unused code in be_cmd_vlan_config()

    For putting interface into vlan promiscuous mode, we use be_cmd_rx_filter()
    and not be_cmd_vlan_config(). So remove the "promiscuous" argument from
    be_cmd_vlan_config().

    Signed-off-by: Kalesh AP <kalesh.purayil@emulex.com>
    Signed-off-by: Sathya Perla <sathya.perla@emulex.com>
    Signed-off-by: David S. Miller <davem@davemloft.net>

Signed-off-by: Ivan Vecera <ivecera@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/emulex/benet/be_cmds.c b/drivers/net/ethernet/emulex/benet/be_cmds.c
index c1ebf50..91d44c8 100644
--- a/drivers/net/ethernet/emulex/benet/be_cmds.c
+++ b/drivers/net/ethernet/emulex/benet/be_cmds.c
@@ -1788,7 +1788,7 @@ err:
 
 /* Uses sycnhronous mcc */
 int be_cmd_vlan_config(struct be_adapter *adapter, u32 if_id, u16 *vtag_array,
-         u32 num, bool promiscuous)
+         u32 num)
 {
  struct be_mcc_wrb *wrb;
  struct be_cmd_req_vlan_config *req;
@@ -1808,16 +1808,12 @@ int be_cmd_vlan_config(struct be_adapter *adapter, u32 if_id, u16 *vtag_array,
           wrb, NULL);
 
  req->interface_id = if_id;
- req->promiscuous = promiscuous;
  req->untagged = BE_IF_FLAGS_UNTAGGED & be_if_cap_flags(adapter) ? 1 : 0;
  req->num_vlan = num;
- if (!promiscuous) {
-  memcpy(req->normal_vlan, vtag_array,
-         req->num_vlan * sizeof(vtag_array[0]));
- }
+ memcpy(req->normal_vlan, vtag_array,
+        req->num_vlan * sizeof(vtag_array[0]));
 
  status = be_mcc_notify_wait(adapter);
-
 err:
  spin_unlock_bh(&adapter->mcc_lock);
  return status;
diff --git a/drivers/net/ethernet/emulex/benet/be_cmds.h b/drivers/net/ethernet/emulex/benet/be_cmds.h
index f4ea6ba..451f313 100644
--- a/drivers/net/ethernet/emulex/benet/be_cmds.h
+++ b/drivers/net/ethernet/emulex/benet/be_cmds.h
@@ -2060,7 +2060,7 @@ int be_cmd_get_fw_ver(struct be_adapter *adapter, char *fw_ver,
         char *fw_on_flash);
 int be_cmd_modify_eqd(struct be_adapter *adapter, struct be_set_eqd *, int num);
 int be_cmd_vlan_config(struct be_adapter *adapter, u32 if_id, u16 *vtag_array,
-         u32 num, bool promiscuous);
+         u32 num);
 int be_cmd_rx_filter(struct be_adapter *adapter, u32 flags, u32 status);
 int be_cmd_set_flow_control(struct be_adapter *adapter, u32 tx_fc, u32 rx_fc);
 int be_cmd_get_flow_control(struct be_adapter *adapter, u32 *tx_fc, u32 *rx_fc);
diff --git a/drivers/net/ethernet/emulex/benet/be_main.c b/drivers/net/ethernet/emulex/benet/be_main.c
index 56f7b5b..8a7249e 100644
--- a/drivers/net/ethernet/emulex/benet/be_main.c
+++ b/drivers/net/ethernet/emulex/benet/be_main.c
@@ -1108,8 +1108,7 @@ static int be_vid_config(struct be_adapter *adapter)
  for_each_set_bit(i, adapter->vids, VLAN_N_VID)
   vids[num++] = cpu_to_le16(i);
 
- status = be_cmd_vlan_config(adapter, adapter->if_handle, vids, num, 0);
-
+ status = be_cmd_vlan_config(adapter, adapter->if_handle, vids, num);
  if (status) {
   /* Set to VLAN promisc mode as setting VLAN filter failed */
   if (status == MCC_ADDL_STS_INSUFFICIENT_RESOURCES)
-- 
1.7.1